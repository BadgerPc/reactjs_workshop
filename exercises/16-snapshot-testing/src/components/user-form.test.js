import React from 'react'
import UserForm from './user-form'
import renderer from 'react-test-renderer'

test('Renders with empty input + disabled button', () => {
  const component = renderer.create(<UserForm />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Button gets enabled after adding input', () => {
  const dummyFunction = () => {}
  const component = renderer.create(<UserForm onSubmit={dummyFunction} />)
  let tree = component.toJSON()

  const input = tree.children[0]
  input.props.onChange({ target: { value: 'a' } })

  const button = tree.children[1]
  button.props.onClick()

  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Renders with initial user + enabled button', () => {})

test('Button gets disabled afer removing input', () => {})
