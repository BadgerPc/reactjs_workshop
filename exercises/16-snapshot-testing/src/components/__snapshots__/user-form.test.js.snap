// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button gets enabled after adding input 1`] = `
<div
  className="user-form"
>
  <input
    defaultValue="a"
    onChange={[Function]}
    placeholder="github username"
    type="text"
  />
  <button
    disabled={false}
    onClick={[Function]}
  >
    See profile
  </button>
</div>
`;

exports[`Renders with empty input + disabled button 1`] = `
<div
  className="user-form"
>
  <input
    defaultValue=""
    onChange={[Function]}
    placeholder="github username"
    type="text"
  />
  <button
    disabled={true}
    onClick={[Function]}
  >
    See profile
  </button>
</div>
`;
